src/meta/Base.ts

```diff
@@ -0,0 +1,23 @@
+import Map from '@dojo/shim/Map';
+import { WidgetMeta, WidgetMetaProperties } from '../interfaces';
+
+export class Base implements WidgetMeta {
+	protected nodes: Map<string, HTMLElement>;
+
+	constructor(properties: WidgetMetaProperties) {
+		this.nodes = properties.nodes;
+		this.invalidate = properties.invalidate;
```

Comment:
So the plan was to allow people to write their own meta class without having to subclass anything. But that's not absolutely necessary and I can move everything into the base class and force people to subclass it.